integrations/graal/mp-native-image-extension/src/main/java/io/helidon/integrations/graal/mp/nativeimage/extension/WeldFeature.java

```diff
@@ -168,11 +168,15 @@ private void initializeProxy(DuringSetupAccess access,
 
     private void iterateBeans(BeanManagerImpl bm,
                               ClientProxyProvider cpp,
-                              Set<Class<?>> processed,
+                              Set<BeanId> processed,
                               Collection<Bean<?>> beans) {
         for (Bean<?> bean : beans) {
-            Class<?> clazz = bean.getBeanClass();
-            if (!processed.add(clazz)) {
+            Set<Type> beanTypes = bean.getTypes();
+
+            BeanId id = new BeanId(bean.getBeanClass(), beanTypes);
```

Comment:
Depending on how proper you want to be, you may want to see if the `bean` is `PassivationCapable` and if so use the return value of its `getId()` method.

Reply:
I think we can live with a certain amount of improper, unless this would fail for some cases.